### PR TITLE
Upgraded robocop version to 0.45.0, to match Hound's version

### PR DESCRIPTION
**Pull request by aoby**
Similar to https://github.com/code-dot-org/code-dot-org/pull/10684

I don't quite understand how / when Hound CI upgrades its rubocop version and I am following up with them over email. Ideally we can lock the version to match our bundle version, or at least be notified when they change it. Anyway Hound [updated](https://github.com/houndci/linters/pull/79) their rubocop version last month, and suddenly today that new version was applied and [caught](https://github.com/code-dot-org/code-dot-org/pull/12513) some violations from the new version that were not violations yesterday in CI, and still are not in our repo.

Steps:

 1. Change the rubocop version specified in the Gemfile.
 2. Run bundle update --source rubocop to update Gemfile.lock.
 3. Run bundle exec rubocop --auto-gen-config --exclude-limit 0 to regenerate .rubocop_todo.yml (the new version of rubocop has a different set of rules).
 4. Run rake lint:ruby to verify, no errors.

NOTE - the previous instructions said "import the new .rubocop_hound.yml from Hound's repo (no change)." but I couldn't find this file in hound.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -214,31 +244,36 @@ Style/Next:
...
-# Offense count: 49
+# Offense count: 51
 # Cop supports --auto-correct.
 Style/RedundantParentheses:
   Enabled: false
 
+# Offense count: 6
+# Cop supports --auto-correct.
+Style/RedundantSelf:
+  Enabled: false
```
Comment (ashercodeorg) [inline .rubocop_todo.yml:256]: This was previously fixed. Please add an explicit TODO (feel welcome to assign to me) to eliminate these.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,52 +1,77 @@
...
+# Offense count: 48
 # Cop supports --auto-correct.
 # Configuration parameters: AllowUnusedKeywordArguments, IgnoreEmptyMethods.
 Lint/UnusedMethodArgument:
   Enabled: false
 
-# Manually added due to internal error in the rubocop rule that needs to be investigated further.
-# To repro and see stack trace, enable this and run: bundle exec rubocop bin/cron/deliver_poste_messages -d
-Lint/ShadowedException:
-  Enabled: false
```
Comment (ashercodeorg) [inline .rubocop_todo.yml:37]: Did this resolve itself, presumably, with the new version?

**Comment by ashercodeorg** [comment]
LGTM. Thanks!

File: .rubocop_todo.yml (hunk 1 above)
Comment (aoby) [reply .rubocop_todo.yml:256]: I wonder how these slipped through? Maybe they tightened the rule?

File: .rubocop_todo.yml (hunk 2 above)
Comment (aoby) [reply .rubocop_todo.yml:37]: That was the one with the internal failure that was crashing rubocop. They must have fixed it.

File: .rubocop_todo.yml (hunk 1 above)
Comment (aoby) [reply .rubocop_todo.yml:256]: Note all the violations are in old code that hasn't changed:

```
Offenses:

dashboard/test/helpers/levels_helper_test.rb:278:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha
                  ^^^^^^^^^^^^^^^^
dashboard/test/helpers/levels_helper_test.rb:289:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha
                  ^^^^^^^^^^^^^^^^
dashboard/test/helpers/levels_helper_test.rb:296:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha
                  ^^^^^^^^^^^^^^^^
dashboard/test/helpers/levels_helper_test.rb:306:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha ha
                  ^^^^^^^^^^^^^^^^
dashboard/test/helpers/levels_helper_test.rb:317:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha ha
                  ^^^^^^^^^^^^^^^^
dashboard/test/helpers/levels_helper_test.rb:324:19: C: Style/RedundantSelf: Redundant self detected.
    app_options = self.app_options # ha ha
                  ^^^^^^^^^^^^^^^^
```

**Pull request merged**